### ISSUE TITLE
Poller.Remove() concurrency bug

### ISSUE THREAD

**Issue by ashmrtn** [zeromq/goczmq#299]
There appears to be an unsafe compiler optimization that causes a panic due to an out of bounds exception when accessing the `Poller.socks` slice in `Poller.Remove()` from multiple go routines, even if the calls to `Add()` and `Remove()` are properly synchronized with a mutex. This occurs even if `Poller.Wait()` is never called, meaning the only accesses to the `socks` slice are from `Add()` and `Remove()`. This error consistently occurs with go version `go1.16.5 linux/amd64` and goczmq `v4.1.0`, but I have not tried to reproduce on other versions/OSes.

I know this project doesn't control the compiler, but it would be nice if this code doesn't cause a panic even if the compiler does weird stuff.

A minimal reproducibler can be found at:
https://gist.github.com/ashmrtn/5594f7c2c8c53ba5c97ca83f719c25ed

When running, the failure is reported similar to the following (index position and offending goroutine vary):
```
builder@c284ad3abe64:/go/src/test$ ./test
panic: runtime error: index out of range [9] with length 9

goroutine 9 [running]:
github.com/zeromq/goczmq.(*Poller).Remove(...)
        /go/pkg/mod/github.com/zeromq/goczmq@v4.1.0+incompatible/poller.go:70
main.socketsss(0xc000076020, 0xc00001a078)
        /go/src/test/test.go:28 +0x37f
main.main.func1(0xc000076020, 0xc00001a078, 0xc00001a080)
        /go/src/test/test.go:46 +0x35
created by main.main
        /go/src/test/test.go:45 +0xee
```

A suggested fix for this is to change all lines in `Poller.Remove()` to use `len(p.socks)` directly instead of placing that value in the local variable `numItems`. Doing so on my local version of goczmq allows the given reproducer to run without any problems for me

**Comment by ashmrtn** [comment]
scratch that, it's not actually a concurrency bug or a compiler bug. After more debugging turns out that it's just forgetting to return from `Remove()` after actually removing the socket from `p.socks`. That combined with `newItems` means the function will fail if anything but the last socket is removed from the poller

**Comment by ashmrtn** [comment]
I see this is actually solved in e2dbde6e770ac1a6cbba374590374fc62f900a39. It would be super great if a new tagged version could be made so that `go get` pulls in fixes like this automatically
